rasa/nlu/extractors/LookupEntityExtractor.py

```diff
@@ -0,0 +1,129 @@
+            common_utils.raise_warning(
+                f"Can't extract Lookup Entities, Please configure the entities and file path for 'CustomEntityExtractor' in the config.yml."
+            )
+
+        # check if entities are configured with their respective file path
+        elif component_config["entities"] is not None and component_config["files_path"] is not None:
+            if len(component_config["entities"]) != len(component_config["files_path"]):
+                self.component_config["files_path"] = None
+                common_utils.raise_warning(
+                    f"Can't extract Lookup Entities, Make sure you have configured the entities and their respective file path properly"
+                )
+        # check if the given file path exists
+        elif component_config["files_path"] is not None:
+            for i in range(len(component_config["files_path"])):
+                file_path = component_config["files_path"][i]
+                print("file_path: ", file_path)
+                # try:
+                if os.path.isfile(file_path):
+                    pass
+                else:
+                    self.component_config["files_path"] = None
+                    common_utils.raise_warning(
+                        f"Can't extract Lookup Entities, File '{file_path}' does not exist."
+                    )
+                    break
+
+    def add_extractor_name(
+        self, entities: List[Dict[Text, Any]]
+    ) -> List[Dict[Text, Any]]:
+        """
+        Adds the Extractor name to the Message class during the prediction.
+        """
+        for entity in entities:
+            entity["extractor"] = self.name
+        return entities
+
+    @staticmethod
+    def _parse_entities(self, text: Text) -> List[Dict[Text, Any]]:
+        """
+        pass the user input to  extract the entities
+
+        """
+
+        user_input = text
+        entities = self.component_config["entities"]
+        files_path = self.component_config["files_path"]
+
+        if files_path is not None and entities is not None:
+            if len(entities) != len(files_path):
+                return []
+            else:
+                results = self._parse_all_entities(
+                    user_input, entities, files_path)
+                return results
+        else:
+            return []
+
+    @staticmethod
+    def _parse_all_entities(user_input: str, entities: list, file_path: list) -> List[Dict[Text, Any]]:
+        """
+        This method does the actual entity extraction work.\n
+        So here I am running the loop over the list of data in the text file\n
+        and check whether it exists in the user's message
+        """
+        results = []
+        for i in range(0, len(entities)):
+            f = open(file_path[i], "r")
+            examples = f.readlines()
+
+            for example in examples:
+                if example.lower().strip() in user_input.lower():
```

Comment:
You seem to lower case everything here. I can certainly see that there is merit to this but I wonder if this is the behavior you'd want in all scenarios. It might be better to have this be a configuration. @tabergma you might also have an opinion on this.

Reply:
If in the Lookup file, if there is any value which is lower case or user query has both the things, so to avoid the casing issue, I preferred lower case everything here. let me know if this behavior increases the code processing time.

Reply:
I'll resolve this conversation since tanja replied with a more indepth correspondence below.